File: src/javascript/swing/ButtonGroup.java
package javascript.swing;

import def.js.Date;
import static simulation.js.$Globals.parseInt;

/**
 * The javax.swing.ButtonGroup clone
 *
 * @author gianpiero.diblasi
 */
public class ButtonGroup {

  private final String name = "ButtonGroup_" + new Date().getTime() + "_" + parseInt(1000 * Math.random());

  /**
   * Adds a button to the group
   *
   * @param button The button
   */
  public void add(JSRadioButton button) {
    button.element.querySelector("[type=radio]").setAttribute("name", this.name);
  }
}

File: transpile/javascript/swing/ButtonGroup.js
/**
 * The javax.swing.ButtonGroup clone
 *
 * @author gianpiero.diblasi
 */
class ButtonGroup {

   name = "ButtonGroup_" + new Date().getTime() + "_" + parseInt(1000 * Math.random());

   count = 0;

  /**
   * Clone of javax.swing.ButtonGroup.add
   *
   * @param button The button
   */
   add(button) {
    button.setChildAttributeByQuery("[type=radio]", "name", this.name);
    this.count++;
  }

  /**
   * Clone of javax.swing.ButtonGroup.getButtonCount
   *
   * @return the button count
   */
   getButtonCount() {
    return this.count;
  }
}
